src/ServicePulse.Host/app/modules/configuration/js/license/license.controller.js

```diff
@@ -14,12 +14,52 @@
 
         function mapLicenseToVm(license) {
             vm.licenseType = license.license_type;
+            vm.licenseEdition = license.edition ? ", " + license.edition : "";
             vm.scInstanceName = license.instance_name;
-            vm.formattedExpirationDate = formatter.formatDate(license.expiration_date);
-            vm.expirationDaysLeft = formatter.getDayDiff(license.expiration_date);
-            vm.formattedUpgradeProtectionExpiration =
-                formatter.formatDate(license.upgrade_protection_expiration);
-            vm.upgradeDaysLeft = formatter.getDayDiff(license.upgrade_protection_expiration);
+            vm.license_status = license.license_status;
+            if (license.expiration_date) {
+                vm.formattedExpirationDate = new Date(license.expiration_date.replace("Z", "")).toLocaleDateString();
+            }
+            if (license.upgrade_protection_expiration) {
+
+                vm.formattedUpgradeProtectionExpiration =
+                    new Date(license.upgrade_protection_expiration.replace("Z", "")).toLocaleDateString();
+            }
+
+            var status = license.license_status;
+            vm.isTrialLicense = license.license_type === "Trial";
+            vm.isUpgradeProtectionLicense = license.upgrade_protection_expiration !== "";
+            vm.isSubscriptionLicense = license.expiration_date !== "" && !vm.isTrialLicense;
+            vm.isExpiring = licenseMatches(status,
+                "ValidWithExpiringSubscription",
+                "ValidWithExpiringTrial",
+                "ValidWithExpiringUpgradeProtection");
+            vm.isExpired = licenseMatches(status,
+                "InvalidDueToExpiredTrial",
+                "InvalidDueToExpiredSubscription",
+                "ValidWithExpiredUpgradeProtection",
+                "InvalidDueToExpiredUpgradeProtection");
+            vm.isValid = !licenseMatches(status,
+                "InvalidDueToExpiredTrial",
+                "InvalidDueToExpiredSubscription",
+                "InvalidDueToExpiredUpgradeProtection");
+
+            if (!vm.isValid) {
+                vm.expiredWarningType = "danger";
+            } else if (vm.isExpiring || (vm.isExpired && vm.isUpgradeProtectionLicense)) {
+                vm.expiredWarningType = "warning";
+            }
+
+            vm.expirationDaysLeft = vm.isExpired ? " - expired" : formatter.getDayDiffFromToday(license.expiration_date);
+            vm.upgradeDaysLeft =
+                vm.isExpired ? " - expired" : formatter.getDayDiffFromToday(license.upgrade_protection_expiration);
+        }
+
+        function licenseMatches(status) {
+            for (var i = 1; i < arguments.length; i++) {
```

Comment:
Starting from index 1? What is this, python? 

We can use destructuring to extract the first argument from the rest and then use a regular `filter` method instead of doing a for iteration. (I guess it's personal style, but it feels better to me)

```suggestion
function licenseMatches() {
    const [status, ...matches] = arguments;
    const matched = matches.find(m => m === status);
    
    return typeof matched !== "undefined";
}
```